crashsimilarity/utils.py

```diff
@@ -17,6 +17,37 @@ def read_files(file_names, open_file_function=smart_open):
                 yield line if isinstance(line, str) else line.decode('utf8')
 
 
+class StackTraceProcessor(object):  # just a namespace, actually
+    @staticmethod
+    def _should_skip(stack_trace):
+        """Exclude stack traces without symbols"""
+        return any(call in stack_trace for call in ['xul.dll@', 'XUL@', 'libxul.so@'])
+
+    @staticmethod
+    def _preprocess(stack_trace, take=None):
+        def clean(func):
+            func = func.lower().replace('\n', '')
+            return func[:func.index('@0x') + 3] if '@0x' in func else func
+
+        traces = [clean(f).strip() for f in stack_trace.split(' | ')]
+        if take:
+            traces = traces[:take]
+        return traces
+
+    @staticmethod
+    def process(stream):
```

Comment:
I don't know if this method belongs here, as this is not a transformation on stack trace/function.
This is more of a IO function.